cypress/integration/regression/main-table-regression.spec.ts

```diff
@@ -0,0 +1,226 @@
+    datasetPage.leftPanel.datasetsListElem
+      .getButtonByText('PGP3140_wgs_panel_hl')
+      .click()
+    datasetPage.leftPanel.leftPanelHeader.checkLabelText('Datasets')
+    cy.wait(500)
+    datasetPage.datasetInfo.openInViewer.click()
+    datasetPage.datasetInfo.mainTable.click()
+    cy.url().should('include', '/ws?ds=PGP3140_wgs_panel_hl')
+  })
+
+  it('should apply preset on a dataset | step 2', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: 20000 })
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.selectPreset.click()
+    cy.intercept('POST', '/app/tab_report').as('loadPreset')
+    mainTablePage.mainTable.preset.getButtonByText(homozygous).click()
+    cy.wait('@loadPreset', { timeout: 20000 })
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+  })
+
+  it('should apply gene to the preset | step 3', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: 20000 })
+    mainTablePage.mainTable.selectPreset.click()
+    cy.intercept('POST', '/app/tab_report').as('loadPreset')
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.preset.getButtonByText(homozygous).click()
+    cy.wait('@loadPreset', { timeout: Timeouts.TwentySecondsTimeout })
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+    mainTablePage.mainTable.addGene.click()
+    mainTablePage.mainTable.searchFilter.type('CHSY1')
+    mainTablePage.mainTable.geneCheckbox.check()
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.numVariants.haveText('Variants: 1')
+  })
+
+  it('should add custom tag to the variant | step 4', () => {
+    mainTablePage.visit(link)
+    cy.intercept('POST', '/app/tab_report').as('dsUpload')
+    cy.wait('@dsUpload', { timeout: Timeouts.TwentySecondsTimeout })
+    cy.intercept('POST', '/app/tab_report').as('loadPreset')
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2592')
+    mainTablePage.mainTable.selectPreset.click()
+    mainTablePage.mainTable.preset.getButtonByText(homozygous).click()
+    cy.wait('@loadPreset', { timeout: Timeouts.TwentySecondsTimeout })
+    //number of variants changes
+    mainTablePage.mainTable.numVariants.haveText('Variants: 2')
+    mainTablePage.mainTable.addGene.click()
+    mainTablePage.mainTable.searchFilter.type('CHSY1')
+    mainTablePage.mainTable.geneCheckbox.check()
+    mainTablePage.mainTable.applyButton.click()
+    mainTablePage.mainTable.numVariants.haveText('Variants: 1')
+    mainTablePage.mainTable.tableRow.getButtonByText('p.Y434=').click()
+    variantDrawerPage.variantDrawer.addTag.eq(1).click()
+    variantDrawerPage.variantDrawer.tagInput.type('Custom_tag')
```

Comment:
I'd suggest to create `TestData` class and generate test data there in static methods like `getTagName()`. You can use `faker` library for that.